src/Microsoft.AspNetCore.SignalR.Client/ConnectionFactory.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.IO.Pipelines;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.SignalR.Client
+{
+    // REVIEW: Split into Microsoft.AspNetCore.Sockets.Client?
+    public class ConnectionFactory
+    {
+        private readonly HttpClient _httpClient;
+        private readonly ILoggerFactory _loggerFactory;
+        private readonly PipelineFactory _pipelineFactory;
+        private readonly ILogger _logger;
+
+        public ConnectionFactory(HttpClient httpClient, PipelineFactory pipelineFactory, ILoggerFactory loggerFactory)
```

Comment:
The most common use of Pipeline Factory is to create a pair of `IPipelineConnection` objects, representing either end of a pipeline. I wonder if we should put that on an interface or something and use that instead. That's all we need here, a pair of pipelines representing input/output.